### PR TITLE
feat: merge attemptDeadlineSeconds and timeoutSeconds option for v2 scheduled functions

### PR DESCRIPTION
**Pull request by taeold**
We are removing explicit control for users to set `attemptDeadlineSeconds`. This is because we feel that `attemptDeadlineSeconds` and `timeoutSeconds` for the function is deeply coupled - there isn't a valid use case we can think of that requires the two values to be different.

Per Cloud Scheduler docs:

> The deadline for job attempts. If the request handler does not respond by this deadline then the request is cancelled and the attempt is marked as a DEADLINE_EXCEEDED failure. The failed attempt can be viewed in execution logs. Cloud Scheduler will retry the job according to the [RetryConfig](https://docs.cloud.google.com/scheduler/docs/reference/rest/v1beta1/projects.locations.jobs#RetryConfig).
>
> The default and the allowed values depend on the type of target:
>
> For [HTTP targets](https://docs.cloud.google.com/scheduler/docs/reference/rest/v1beta1/projects.locations.jobs#Job.FIELDS.http_target), the default is 3 minutes. The deadline must be in the interval [15 seconds, 30 minutes]. 

Effectively, we believe function timeout ~= attemptDeadline (ideally attemptDeadline should be a second or two greater than function timeout, but we leave that out as edge case for now)

### CODE REVIEW

File: tests/test_scheduler_fn.py [hunk 1]
```diff
@@ -47,6 +47,18 @@ def test_on_schedule_decorator(self):
...
+        Tests that attemptDeadlineSeconds is set to timeoutSeconds.
+        """
+        decorated_func = scheduler_fn.on_schedule(
+            schedule="* * * * *",
+            timeout_sec=120,
+        )(Mock(__name__="example_func"))
+        endpoint = decorated_func.__firebase_endpoint__
+        self.assertEqual(endpoint.timeoutSeconds, 120)
+        self.assertEqual(endpoint.scheduleTrigger.get("attemptDeadlineSeconds"), 120)
+
```
[the remark this answers is not in the document]
Comment (taeold) [reply tests/test_scheduler_fn.py:61]: disagree. scheduler having a default timeout of 3m. functions have default timeout of 1m. when scheudler timeout > function timeout, we don't expect things to go wrong.
